Add the MINLP solver MAiNGO to the documentation

MAiNGO-github opened the pull request:
We have recently built a wrapper around our open-source global MINLP solver MAiNGO, which is is written in C++.
The wrapper initially took a lot of inspiration from the Baron.jl repository.

Looking at the [checklist](https://jump.dev/JuMP.jl/stable/developers/checklists/#Adding-a-new-solver-to-the-documentation), the conditions for including the solver in the documentation should be fulfilled.

We would be very excited to be included in the list of compatible solvers.

https://github.com/MAiNGO-github/MAiNGO.jl

## Basic

 - [x] Check that the solver is a registered Julia package
 - [x] Check that the solver supports the long-term support release of Julia
 - [x] Check that the solver has a MathOptInterface wrapper
 - [x] Check that the tests call `MOI.Test.runtests`. Some test excludes are
       permissible, but the reason for skipping a particular test should be
       documented.
 - [x] Check that the README and/or documentation provides an example of how to
       use the solver with JuMP

## Documentation

 - [x] Add a new row to the table in `docs/src/installation.md`

## Optional

 - [x] Add package metadata to `docs/packages.toml`

odow commented:
Nice! Let me take a look

odow reviewed (changes requested):
I'm going to mark this as blocked by https://github.com/MAiNGO-github/MAiNGO.jl/issues/2, but it looks nice!

Pleased you figured out the Yggdrasil stuff :smile:

MAiNGO-github commented:
@odow 
I added the Github actions.
I also incorporated a recent improvement from Baron.jl

There are were spellchecking warnings. I fixed those that were correct. Some are caused by how our README is set up.
In case this is blocking or annoying, I would suggest not adding to docs/packages.toml, but only to the overview table.
